alacritty/src/config/window.rs

```diff
@@ -74,11 +74,30 @@ impl Default for WindowConfig {
 impl WindowConfig {
     #[inline]
     pub fn dimensions(&self) -> Option<Dimensions> {
-        if self.dimensions.columns.0 != 0
-            && self.dimensions.lines != 0
-            && self.startup_mode != StartupMode::Maximized
-        {
+        let (lines, columns) = (self.dimensions.lines, self.dimensions.columns.0);
+        let (lines_is_non_zero, columns_is_non_zero) = (lines != 0, columns != 0);
+
+        if lines_is_non_zero && columns_is_non_zero {
+            // return dimensions if both `lines` and `columns` is non-zero
             Some(self.dimensions)
+        } else if lines_is_non_zero || columns_is_non_zero {
+            // Warn if either `columns` or `lines` is non-zero
```

Comment:
```suggestion
            // Warn if either `columns` or `lines` is non-zero.
            
```

Extra newline since this comment isn't directly attached to the block following it.